index.js

```diff
@@ -1,6 +1,80 @@
 /* jshint node: true */
 'use strict';
 
+const Funnel = require('broccoli-funnel');
+const chalk  = require('chalk');
+
+// We import these to merge the instrumentation options into the base classes
+// we do this instead of overriding the core commands because other high profile
+// addons such as ember-cordova, ember-cli-nwjs, and ember-electron have had to override
+const serveCommand = require('ember-cli/lib/commands/serve');
+const buildCommand = require('ember-cli/lib/commands/build');
+const testCommand = require('ember-cli/lib/commands/test');
+
+function patchCommand(command, newOption) {
+  command.prototype.availableOptions.push(newOption);
+}
+
+function assert(msg, test) {
+  if (!test) {
+    throw new Error(msg);
+  }
+}
+
 module.exports = {
-  name: 'ember-heimdall'
+  name: 'ember-heimdall',
+
+  _isInstrumented: null,
+  isInstrumented: function() {
+    if (this._isInstrumented !== null) {
+      return this._isInstrumented;
+    }
+
+    let INSTRUMENT_HEIMDALL = false;
```

Comment:
Might be simpler to do:

    let INSTRUMENT_HEIMDALL = process.argv.indexOf('--instrument') > -1;